launchdarkly-server-sdk/src/events/event.rs

```diff
@@ -95,24 +95,135 @@ impl BaseEvent {
 }
 
-#[derive(Clone, Debug, Serialize)]
-#[serde(rename_all = "camelCase")]
 /// MigrationOpEventData is generated through the migration op tracker provided through the SDK.
+#[derive(Clone, Debug)]
 pub struct MigrationOpEvent {
-    #[serde(flatten)]
     pub(crate) base: BaseEvent,
     pub(crate) key: String,
     pub(crate) version: Option<u64>,
     pub(crate) operation: Operation,
     pub(crate) default_stage: Stage,
     pub(crate) evaluation: Detail<Stage>,
     pub(crate) invoked: HashSet<Origin>,
-    #[serde(skip_serializing_if = "Option::is_none")]
     pub(crate) consistency_check: Option<bool>,
+    pub(crate) consistency_check_ratio: Option<u64>,
     pub(crate) errors: HashSet<Origin>,
     pub(crate) latency: HashMap<Origin, Duration>,
 }
 
+impl Serialize for MigrationOpEvent {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_struct("MigrationOpEvent", 10)?;
+        state.serialize_field("kind", "migration_op")?;
+        state.serialize_field("creationDate", &self.base.creation_date)?;
+        state.serialize_field("contextKeys", &self.base.context.context_keys())?;
+        state.serialize_field("operation", &self.operation)?;
+
+        let evaluation = MigrationOpEvaluation {
+            key: self.key.clone(),
+            value: self.evaluation.value,
+            // QUESTION: In the ruby implementation, this can be nil. Why not here?
+            reason: self.evaluation.reason.clone(),
+            variation_index: self.evaluation.variation_index,
+            version: self.version,
+        };
+        state.serialize_field("evaluation", &evaluation)?;
+        state.serialize_field("default", &self.default_stage)?;
+
+        // TODO: Add sampling here if it is set and not 1
+
+        let mut measurements = vec![];
+        if self.invoked.is_empty() {
+            measurements.push(MigrationOpMeasurement::Invoked(&self.invoked));
+        }
+
+        // TODO: There is something here to do with consistency check ratio
+        if let Some(consistency_check) = self.consistency_check {
+            measurements.push(MigrationOpMeasurement::ConsistencyCheck(
+                consistency_check,
+                self.consistency_check_ratio,
+            ));
+        }
+
+        if !self.errors.is_empty() {
+            measurements.push(MigrationOpMeasurement::Errors(&self.errors));
+        }
+
+        if !self.latency.is_empty() {
+            measurements.push(MigrationOpMeasurement::Latency(&self.latency));
+        }
+
+        state.end()
+    }
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct MigrationOpEvaluation {
+    pub key: String,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub value: Option<Stage>,
+
+    pub reason: Reason,
+
+    #[serde(rename = "variation", skip_serializing_if = "Option::is_none")]
+    pub variation_index: Option<VariationIndex>,
```

Comment:
Too bad Detail<T> encodes as variationIndex instead of variation. Otherwise, we could rely on Detail's serialization directly. 😬